auth_mellon_util.c

```diff
@@ -1540,7 +1612,11 @@ bool am_validate_paos_header(request_rec *r, const char *header)
         goto cleanup;
     }
 
-    /* Split the header on the semicolon character */
+    /*
+     * Split the header on the semicolon character.
+     * There should be 2 items, the first is the version delclaration
+     * and the second is the service and it's options.
+     */
     semicolon_tokens = g_strsplit(header, ";", 0);
```

Comment:
Considering the complexity of this header now, I am wondering how safe/unsafe this method of parsing it is. Before it was very strict in what it accepted, but now it tries to deal with all sorts of extra flags/options, but it still uses `g_strsplit()` for parsing, both here, and now also `service_tokens` further down.

Some reasssuring words about how commas and semicolons can never occur as part of neither the version field nor the options would be appreciated. :smiley:

Just to be clear, I am thinking of headers like:

```
PAOS: ver="urn:liberty:paos:2003-08"; "somerandomoption;param=42"
```

I have no idea if that sort of thing is supposed to be possible, but the fact that they allow (or encourage?) the use of double-quotes makes me wonder about it.